extensions/blocks/rating-star/rating-meta.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Utilities for the rating block.
+ *
+ * @package Jetpack
+ */
+
+/**
+ * Returns the low fidelity symbol for the block.
+ *
+ * @return string
+ */
+function jetpack_rating_meta_get_symbol_low_fidelity() {
+	return '⭐';
+}
+
+/**
+ * Returns the high fidelity symbol for the block.
+ *
+ * @param array   $attributes Array containing the business hours block attributes.
+ * @param integer $pos Value to render whole and half symbols.
+ * @return string
+ */
+function jetpack_rating_meta_get_symbol_high_fidelity( $attributes, $pos ) {
+	$classname_whole = ( $attributes['rating'] >= ( $pos - 0.5 ) ) ? '' : 'is-rating-unfilled';
+	$classname_half  = ( $attributes['rating'] >= $pos ) ? '' : 'is-rating-unfilled';
+	$color           = empty( $attributes['color'] ) ? 'currentColor' : esc_attr( $attributes['color'] );
+
+	return <<<ELO
+<span>
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
+	<path class="{$classname_whole}" fill="{$color}" stroke="{$color}" d="M12,17.3l6.2,3.7l-1.6-7L22,9.2l-7.2-0.6L12,2L9.2,8.6L2,9.2L7.5,14l-1.6,7L12,17.3z" />
+</svg>
+</span>
+<span>
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
+	<path class="{$classname_half}" fill="{$color}" stroke="{$color}" d="M12,17.3l6.2,3.7l-1.6-7L22,9.2l-7.2-0.6L12,2L9.2,8.6L2,9.2L7.5,14l-1.6,7L12,17.3z" />
+</svg>
+</span>
+ELO;
+
+}
+
+/**
+ * Returns the symbol for the block.
+ *
+ * @param array $attributes Array containing the business hours block attributes.
+ *
+ * @return string
+ */
+function jetpack_rating_meta_get_symbols( $attributes ) {
+	// Output SVGs for high fidelity contexts, then color them according to rating.
+	// These are hidden by default, then unhid when CSS loads.
+	$symbols_hifi = array();
+	for ( $pos = 1; $pos <= $attributes['maxRating']; $pos++ ) {
+		$symbols_hifi[] = '<span style="display: none;">' . jetpack_rating_meta_get_symbol_high_fidelity( $attributes, $pos ) . '</span>';
+	}
+
+	// Output fallback symbols for low fidelity contexts, like AMP,
+	// where CSS is not loaded so the high-fidelity symbols won't be rendered.
+	$symbols_lofi = '';
+	for ( $i = 0; $i < $attributes['rating']; $i++ ) {
+		$symbols_lofi .= jetpack_rating_meta_get_symbol_low_fidelity();
+	}
+
+	return '<p>' . $symbols_lofi . '</p>' . implode( $symbols_hifi );
```

Comment:
I wonder if we should add more to that markup to make the block output more accessible to screen readers? I'm no pro there, I am not quite sure how this would get read. VoiceOver does not seem to read anything in my tests.

Reply:
I'd let @jasmussen chime in here, but I think it is ok as it is as per these changes (see issue description) https://github.com/Automattic/wpcom-blocks/pull/123

Reply:
Things should actually be a fair bit improved here compared to before. I don't know that the past version was very screen-reader accessible at all (😔) but the new one at least reads aloud the stars as "White Medium Star, White Medium Star, White Medium Star" (for a 2.5 or 3 star rating).

Here's a GIF I recorded trying to learn how the emoji are announced:

![emoji](https://user-images.githubusercontent.com/1204802/69533701-947e0380-0f78-11ea-9330-2aae25ce9dcb.gif)

In this one, the 💲 emoji is announced as "Dollar Sign" in comparison to $ which is announced as "dollar". 

I would welcome additional accessibility improvements; I wonder if we could output literal text instead of literal emoji that perhaps read out something like "2.5 White Medium Star rating" or something. 

I would suggest that to be a nice future update to make, but given the improvement over what's there today, I think we should ship this as an iterative improvement.

Reply:
That's interesting, on my end VoiceOver didn't pick up anything when inside a post, but I must have been doing something wrong. I get simple "yellow five-pointed star" indicators when loading our markup on its own:
https://jeherve.keybase.pub/testing-rating.html

I think we could add some content to wrap the whole rating to mention what it is, but I agree that it's not really a blocker for now.

Reply:
Logged this here:
https://github.com/Automattic/jetpack/issues/14120